pom.xml

```diff
@@ -32,22 +32,24 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <!-- Dependency versions -->
-        <groovy.version>2.4.17</groovy.version>
-        <junit.version>4.13.1</junit.version>
+        <groovy.version>3.0.7</groovy.version>
+        <junit.version>5.7.0</junit.version>
         <jenkins-pipeline-unit.version>1.8</jenkins-pipeline-unit.version>
-        <groovy-eclipse-compiler.version>3.4.0-01</groovy-eclipse-compiler.version>
+        <groovy-eclipse-compiler.version>3.6.0-03</groovy-eclipse-compiler.version>
     </properties>
 
     <dependencies>
         <dependency>
             <groupId>org.codehaus.groovy</groupId>
             <artifactId>groovy-all</artifactId>
             <version>${groovy.version}</version>
+            <!-- Type pom replaces the über "groovy-all.jar", for version 3+ -->
+            <type>pom</type>
```

Comment:
do we still need this groovy dependency? or should we be specifying just the actual groovy jar? seems weird to depend on a pom artifact?

Reply:
I have no idea if it is required or not. On 1 side it might be right as this dependency is not mentionned in https://github.com/jenkinsci/JenkinsPipelineUnit#add-to-your-project-as-test-dependency .
On the other, my knowledge is not sufficient enough how to make this work without this 😱 

It looks like that the `groovy-all` (e.g. the Eclipse Groovy Compiler) is not the only solution: https://docs.groovy-lang.org/latest/html/documentation/tools-groovyc.html#section-groovyeclipse . But it is recommended when there is a joint Java-Groovy compilation, which should be the case for shared library if I understand correctly the `src/**` usage is that right?
=> If this assumption is valid, then https://github.com/groovy/groovy-eclipse/wiki/Groovy-Eclipse-Maven-plugin is to be used, but there is no mention of `groovy-all` in this doc. So I don't really know 🤷🏾‍♂️

Reply:
=> http://groovy-lang.org/releasenotes/groovy-2.5.html#Groovy2.5releasenotes-Packaging might help

Reply:
I guess if it works